scripts/build-rss.js

```diff
@@ -16,60 +16,63 @@ function clean(s) {
+    rss.channel.pubDate = new Date().toUTCString()
+    rss.channel.generator = 'next.js'
+    rss.channel.item = []
 
-  const feed = {}
-  const rss = {}
-  rss['@version'] = '2.0'
-  rss["@xmlns:atom"] = 'http://www.w3.org/2005/Atom'
-  rss.channel = {}
-  rss.channel.title = title
-  rss.channel.link = `${base}/${outputPath}`
-  rss.channel["atom:link"] = {}
-  rss.channel["atom:link"]["@rel"] = 'self'
-  rss.channel["atom:link"]["@href"] = rss.channel.link
-  rss.channel["atom:link"]["@type"] = 'application/rss+xml'
-  rss.channel.description = desc
-  rss.channel.language = 'en-gb';
-  rss.channel.copyright = 'Made with :love: by the AsyncAPI Initiative.';
-  rss.channel.webMaster = 'info@asyncapi.io (AsyncAPI Initiative)'
-  rss.channel.pubDate = new Date().toUTCString()
-  rss.channel.generator = 'next.js'
-  rss.channel.item = []
-
-  for (let post of posts) {
-    const link = `${base}${post.slug}${tracking}`;
-    const item = { title: post.title, description: clean(post.excerpt), link, category: type, guid: { '@isPermaLink': true, '': link }, pubDate: new Date(post.date).toUTCString() }
-    if (post.cover) {
-      const enclosure = {};
-      enclosure["@url"] = base+post.cover;
-      enclosure["@length"] = 15026; // dummy value, anything works
-      enclosure["@type"] = 'image/jpeg';
-      if (typeof enclosure["@url"] === 'string') {
-        let tmp = enclosure["@url"].toLowerCase();
-        if (tmp.indexOf('.png')>=0) enclosure["@type"] = 'image/png';
-        if (tmp.indexOf('.svg')>=0) enclosure["@type"] = 'image/svg+xml';
-        if (tmp.indexOf('.webp')>=0) enclosure["@type"] = 'image/webp';
+    for (let post of posts) {
+      const link = `${base}${post.slug}${tracking}`;
+      const item = { title: post.title, description: clean(post.excerpt), link, category: type, guid: { '@isPermaLink': true, '': link }, pubDate: new Date(post.date).toUTCString() }
+      if (post.cover) {
+        const enclosure = {};
+        enclosure["@url"] = base + post.cover;
+        enclosure["@length"] = 15026; // dummy value, anything works
+        enclosure["@type"] = 'image/jpeg';
+        if (typeof enclosure["@url"] === 'string') {
+          let tmp = enclosure["@url"].toLowerCase();
+          if (tmp.indexOf('.png') >= 0) enclosure["@type"] = 'image/png';
+          if (tmp.indexOf('.svg') >= 0) enclosure["@type"] = 'image/svg+xml';
+          if (tmp.indexOf('.webp') >= 0) enclosure["@type"] = 'image/webp';
+        }
+        item.enclosure = enclosure;
       }
-      item.enclosure = enclosure;
+      rss.channel.item.push(item)
     }
-    rss.channel.item.push(item)
-  }
 
-  feed.rss = rss
+    feed.rss = rss
 
-  const xml = json2xml.getXml(feed,'@','',2)
-  fs.writeFileSync(`./public/${outputPath}`, xml, 'utf8')
-};
+    const xml = json2xml.getXml(feed, '@', '', 2)
+    fs.writeFileSync(`./public/${outputPath}`, xml, 'utf8')
+  } catch (err) {
+    throw new Error(`Failed to generate RSS feed: ${err.message}`);
```

Comment:
instead of throwing error, can we return error as promise?

@vishvamsinh28 @anshgoyalevil

Reply:
We are processing the script synchronously, so returning a promise wouldn't be appropriate here

Reply:
Catching of error doesn't depend on whether the function is synchronous or asynchronous. It will still return the error, based on the function return type.